packages/snap/src/index.ts

```diff
@@ -33,13 +19,28 @@ export type RequestParams = {
  * @throws If the request method is not valid for this snap, or if the request params are invalid. 
  */
 export const onRpcRequest: OnRpcRequestHandler = async ({ origin, request }) => {
-  const params = request.params as unknown as RequestParams  // TODO improve type safety
+  const params = request.params as unknown as typeof RequestParams   
```

Comment:
I think you should be able to just get rid of this line entirely

Reply:
could you put this in another way? so we just refer to tx directly from the request object, 


so anywhere we are doing params.tx, we would do request.params.tx?

Reply:
yes exactly

Reply:
ah ok, this bring some issues in the IDE , and could mean we would need to add a // ts-ignore for every line cos of the following highlight - let me know what your thoughts are 

`
Property 'tx' does not exist on type 'Json[] | Record<string, Json>'.
  Property 'tx' does not exist on type 'Json[]'.ts(2339)
`

Reply:
try running `yarn && yarn build`

Reply:
The io-ts check automatically tells the compiler the type after it passes